src/callbacks_step/amr_dg1d.jl

```diff
@@ -76,6 +76,114 @@ function refine!(u_ode::AbstractVector, adaptor, mesh::TreeMesh{1},
 
+function refine!(u_ode::AbstractVector, adaptor, mesh::TreeMesh{1},
+                 equations, dg::DGSEM, cache, cache_parabolic,
+                 elements_to_refine)
+    # Return early if there is nothing to do
+    if isempty(elements_to_refine)
+        return
+    end
+
+    # Determine for each existing element whether it needs to be refined
+    needs_refinement = falses(nelements(dg, cache))
+    needs_refinement[elements_to_refine] .= true
+
+    # Retain current solution data
+    old_n_elements = nelements(dg, cache)
+    old_u_ode = copy(u_ode)
+    GC.@preserve old_u_ode begin # OBS! If we don't GC.@preserve old_u_ode, it might be GC'ed
+        old_u = wrap_array(old_u_ode, mesh, equations, dg, cache)
+
+        # Get new list of leaf cells
+        leaf_cell_ids = local_leaf_cells(mesh.tree)
+
+        # re-initialize elements container
+        @unpack elements = cache
+        resize!(elements, length(leaf_cell_ids))
+        init_elements!(elements, leaf_cell_ids, mesh, dg.basis)
+        @assert nelements(dg, cache) > old_n_elements
+
+        @unpack elements, cache_viscous = cache_parabolic
+        resize!(elements, length(leaf_cell_ids))
+        init_elements!(elements, leaf_cell_ids, mesh, dg.basis)
+        @assert nelements(dg, cache_parabolic) > old_n_elements
+
+        resize!(u_ode,
+                nvariables(equations) * nnodes(dg)^ndims(mesh) * nelements(dg, cache))
+        u = wrap_array(u_ode, mesh, equations, dg, cache)
+
+        # Resize parabolic helper variables
+        resize!(cache_viscous,
+                nvariables(equations) * nnodes(dg)^ndims(mesh) * nelements(dg, cache))
+        cache_parabolic.cache_viscous.u_transformed = unsafe_wrap(Array,
+                                                                  pointer(cache_parabolic.cache_viscous._u_transformed),
+                                                                  (nvariables(equations),
+                                                                   nnodes(dg),
+                                                                   nelements(dg, cache)))
+        cache_parabolic.cache_viscous.gradients = unsafe_wrap(Array,
+                                                              pointer(cache_parabolic.cache_viscous._gradients),
+                                                              (nvariables(equations),
+                                                               nnodes(dg),
+                                                               nelements(dg, cache)))
+        cache_parabolic.cache_viscous.flux_viscous = unsafe_wrap(Array,
+                                                                 pointer(cache_parabolic.cache_viscous._flux_viscous),
+                                                                 (nvariables(equations),
+                                                                  nnodes(dg),
+                                                                  nelements(dg, cache)))
```

Comment:
Resize parabolic/viscous helper variables

Reply:
Was meant to point the reviewers to the parts which are different from AMR for hyperbolic terms